apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/ApolloCompiler.kt

```diff
@@ -206,7 +206,8 @@ object ApolloCompiler {
             generatedSchemaName = options.generatedSchemaName,
             flatten = options.flattenModels,
             scalarMapping = options.scalarMapping,
-            generateDataBuilders = options.generateDataBuilders
+            generateDataBuilders = options.generateDataBuilders,
```

Comment:
@martinbonnin How does the `generateDataBuilders` option differ from the ported `generateModelBuilder` option?

Reply:
`generateDataBuilders` generate data based on the schema and not the models. This is especially useful for fragments:

Assuming something like this:

```graphql
query GetHero {
   hero {
      name
      ...droidFragment
   } 
}

fragment droidFragment on Droid {
   name
   primaryFunction
}
```

The code for each solution is going to look like (not sure about the exact details but that should give the high level overview):

 `generateDataBuilders`:
```java
    BuilderFactory factory = BuilderFactory.DEFAULT
    GetHeroQuery.Data data = GetHeroQuery.buildData(
        factory.buildQuery()
            .hero(
                factory.buildDroid()
                    .name("R2-D2")
                    .primaryFunction("translation")
                    .build()
                )
            )
            .build()
      );
```

`generateModelBuilders`:
```java
    GetHeroQuery.Data data = GetHeroQuery.builder()
       .hero(
          Hero.builder()
                .__typename("Droid")
                .name("R2-D2")
                .droidDetails(
                   DroidDetails.builder()
                      .__typename("Droid")
                      .name("R2-D2")
                      .primaryFunction("translation")
                      .build()
                 )
                .build()
       )
       .build()
```

`generateDataBuilders` has some initial overhead (the factory) but will be simpler as more fragments are involved

Reply:
I see, it's a more fluent interface. Is this the right time to update documentation BTW? Or should that be a separate MR?

Reply:
Documentation should go in a separate MR so that we can hold the documentation updates until the version is actually released.

Reply:
Something else about `data builders` is that they work with `FakeResolver` to provide default values for model fields that haven't been set.